Python/Product/Common/Infrastructure/ProcessOutput.cs

```diff
@@ -330,8 +326,37 @@ select connection.LocalEndPoint.Port
                 string.Join("|", env.Select(kv => kv.Key + "=" + Convert.ToBase64String(utf8.GetBytes(kv.Value)))) :
                 "";
 
-            listener.Start();
-            listener.AcceptTcpClientAsync().ContinueWith(t => {
+            TcpListener listener = null;
+            Task<TcpClient> clientTask = null;
+
+            for (int retries = 10; retries >= 0; --retries) {
+                int port = GetFreePort();
+                listener = new TcpListener(IPAddress.Loopback, port);
+                psi.Arguments = port.ToString();
+                try {
+                    listener.Start();
+                } catch (SocketException) {
+                    if (retries == 0) {
+                        throw;
+                    }
+                    continue;
+                }
+                try {
+                    clientTask = listener.AcceptTcpClientAsync();
```

Comment:
are you going to get a SocketException here?  wouldn't that happen only once you've awaited the result?